src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +180,61 @@ public void Initialize(TelemetryConfiguration configuration)
                             portalSender.Send(traceEvent);
                         }
 
+                        // set up heartbeat
+                        this.HeartbeatProvider.Initialize(configuration);
+
                         this.isInitialized = true;
                     }
                 }
             }
         }
-        
+
+        /// <summary>
+        /// Add a new Health Heartbeat property to the payload sent with each heartbeat.
+        /// 
+        /// To update the value of the property you are adding to the health heartbeat, 
+        /// <see cref="DiagnosticsTelemetryModule.SetHealthProperty"/>.
+        /// 
+        /// Note that you cannot add a HealthHeartbeatProperty with a name that already exists in the HealthHeartbeat
+        /// payload, including (but not limited to) the name of SDK-default items.
+        /// 
+        /// </summary>
+        /// <param name="propertyName">Name of the health heartbeat value to add</param>
+        /// <param name="propertyValue">Current value of the health heartbeat value to add</param>
+        /// <param name="isHealthy">Flag indicating whether or not the property represents a healthy value</param>
+        /// <returns>True if the new payload item is successfully added, false otherwise.</returns>
+        public bool AddHealthProperty(string propertyName, string propertyValue, bool isHealthy)
+        {
+            return this.HeartbeatProvider.AddHealthProperty(propertyName, propertyValue, isHealthy);
+        }
+
+        /// <summary>
+        /// Set an updated value into an existing property of the health heartbeat. The propertyName must be non-null and non-empty
+        /// and at least one of the propertyValue and isHealthy parameters must be non-null.
```

Comment:
So how do I remove the property if null is not allowed?

Reply:
In talks with @cijothomas we felt that the payload contents should be *fairly* static after initialization. That is, you add all the properties that will show up in the heartbeat payload up front with no intention to remove them (the payload remains static for heartbeat). From my C++ perspective this would guard against any fragmentation if someone ever abused the properties here, but I’m not 100% confident that is entirely applicable here except for folks that are messing around with the GC. Also, customers that want to add/remove properties have Events, Metrics, and other such telemetry that they can use to track more fluid properties with anyways.